### PR TITLE
[3.2] [iOS] Option to use storyboards for launch screens.

### LINKED ISSUE #39325: iPhone 11 and XR render at the incorrect resolution
**Godot version:**
3.2.1

**OS/device including version:**
iOS 13.5 GLES3

**Issue description:**
iOS export template does not include the launch images that correspond to the iPhone XR and 11.
This makes the phone render at the resolution of 750 x 1334 (iPhone 6 size) instead of 828 x 1792 

Screenshot 2020-06-05 at 17 27 36

left image is when a correct launch image is used and the blurry right image is without

adding the missing launch images that are shown missing here fixes this

Screenshot 2020-06-05 at 17 21 29

but a better fix is to add a LaunchScreen.storyboard file to the supporting files folder and then
link to it in general setting tab in Xcode. Then deleting the LaunchImages from the assets folder

Screenshot 2020-06-05 at 17 19 00

adding the storyboard file also has the bonus of getting rid of the warning that Xcode displays about LaunchImages are now deprecated in iOS 13

I had been wrestling with a problem on my iPhone 11 for weeks trying to get a pixel perfect image that I could scale for different iPhones This fixed a lot of my issues

**Steps to reproduce:**
export an iOS project

### PR DESCRIPTION
**Pull request by naithar**
This PR gives developer an option to use `.storyboard` file instead of simple images for iOS application's launch screen.
Default value for this setting is `false` so it wouldn't break existing projects export.
By default storyboard uses the same values (background, image, scaling) used for Godot's logo screen. But user can specify specific parameters if required.

Should fix #39325 as well as future warnings or errors regarding incorrect launch image sizes.
Also removes current warnings about `LaunchImage` deprecation.

Tested on multiple devices (iPad, and multiple iPhones). 
Without using custom settings for launch screen the result looks the same as Godot's logo in `Main::setup2()` method call.
One thing to note: to correctly test this change, it might be required to delete resulting app and restart iOS device after reexporting godot project, since iOS seems to be caching launch screen images and storyboards.

Currently fully testable only on `3.2`, but `master`'s version of this commit is also ready for PR and it builds on osx and displays new export options.

### CODE REVIEW

File: platform/iphone/export/export.cpp [hunk 1]
```diff
@@ -257,6 +259,13 @@ void EditorExportPlatformIOS::get_export_options(List<ExportOption> *r_options)
 	r_options->push_back(ExportOption(PropertyInfo(Variant::STRING, "optional_icons/spotlight_40x40", PROPERTY_HINT_FILE, "*.png"), "")); // Spotlight
 	r_options->push_back(ExportOption(PropertyInfo(Variant::STRING, "optional_icons/spotlight_80x80", PROPERTY_HINT_FILE, "*.png"), "")); // Spotlight on devices with retina display
 
+	r_options->push_back(ExportOption(PropertyInfo(Variant::BOOL, "storyboard/use_launch_screen_storyboard"), false));
+	r_options->push_back(ExportOption(PropertyInfo(Variant::INT, "storyboard/image_scale_mode", PROPERTY_HINT_ENUM, "Same as Logo,Center,Scale To Fit,Scale To Fill,Scale"), 0));
+	r_options->push_back(ExportOption(PropertyInfo(Variant::STRING, "storyboard/custom_image@2x", PROPERTY_HINT_FILE, "*.png"), ""));
+	r_options->push_back(ExportOption(PropertyInfo(Variant::STRING, "storyboard/custom_image@3x", PROPERTY_HINT_FILE, "*.png"), ""));
```
Comment (bruvzg) [inline platform/iphone/export/export.cpp:265]: Default `splash.png` for the 2x and 3x size have the same 800x600 size, probably it's better to make them actually be 2x and 3x sized.

**Review by bruvzg — APPROVED** [review]

File: platform/iphone/export/export.cpp (hunk 1 above)
Comment (naithar) [reply platform/iphone/export/export.cpp:265]: Using same size for default `splash.png` is intentional. When I tried to use correctly sized default images - splash screen would differ from Godot’s logo on devices using `@3x`, like iPhone XS. I guess it have something to do with the way logo is displayed when it’s not scaling - it uses exact image size without modifying it in any way

**Comment by akien-mga** [comment]
Thanks!

**Comment by oeleo1** [comment]
This is great! Thanks a lot. I tried it and the storyboard option works well. However, after the storyboard shows up, it disappears and I get a black screen for several seconds (custom color is black in the project settings) until the game is loaded and starts. I guess the problem is of the same nature as the one described for Android in [#21824](https://github.com/godotengine/godot/issues/21824#issuecomment-641075070) and the example project attached there exhibits the problem. Maybe you can have a look. Thanks again for the support of storyboards!

**Comment by naithar** [comment]
@oeleo1 well, iOS loading screen is displayed only while application loads dylibs into memory and performs `finishLoading` method. Then it's up to the application to handle what will be displayed.
The reason for logo not being displayed immediately and showing black screen is that it's displayed only on `setup2()` call, while `setup()` and everything else happening between `setup2()` call and `finishLoading` can take a lot of time to finish.
But changing this behavior atm would probably require a proposal getting evaluated - https://github.com/godotengine/godot-proposals/issues/1175.

**Comment by oeleo1** [comment]
Thanks for your input. The referred proposal suggests removing the boot splash which is not what we need. We do need boot splash, it is absolutely essential for big games (200+MB) which take many seconds to load, No one would launch a game and appreciate a black screen for 10 seconds. So the question is whether we can let the storyboard (as a derivative of the boot splash) last until `finishLoading` so that the game can take over immediately. That black emptiness in the middle is nonsense :-) I'd be happy to continue discussing that there, but the topic and purpose of the proposal sounds different. Thanks.

**Comment by andy-noisyduck** [comment]
The intent of the proposal was to avoid the "double splash" of having a Launcher image followed by another Godot splash for projects that don't need a second splash. Whether you require a single splash or not, in both cases we want to keep the Launch Image until something is ready to take over so that we don't get the period of blackness in the middle. Perhaps that should be fleshed out as its own proposal - they are related but not quite the same.

If your app is massive, you probably want to keep the launch image there long enough to load an initial splash scene, and use background loading (e.g. via ResourceInteractiveLoader) to give loading progress whilst the rest of your game starts up.

**Comment by oeleo1** [comment]
You've done an amazing job on this -- I can't tell you how happy I am to see this commit.

Now, I hear you on the background loading, but that's like taking a U-turn on the standard straightforward loading sequence. What we really need is an image while the game loads "as is". And we just can't get it. Please try the attached simplistic project (2 nodes) on iOS which really shows what our problem is. I'll take any solution. Thanks!
[Boot_Splash.zip](https://github.com/godotengine/godot/files/4920079/Boot_Splash.zip)

**Comment by naithar** [comment]
As I said this changes that you want would probably require changes in how iOS platform performs Godot's loading sequence. This would probably need a proposal or an issue :)

**Comment by oeleo1** [comment]
The issue is [#38897](https://github.com/godotengine/godot/issues/38897) which has been closed and redirected to [#21824](https://github.com/godotengine/godot/issues/21824) which is open since 2.5 years.

Not sure what else we need here, except to understand why the boot image doesn't show on the screen on iOS when Godot writes to the system FBO in `setup2()` before loading the project. It shows only after the project is loaded and the user project is started. This is what causes the black emptiness in the middle and it is probably related to app styling, decorelated to the way Godot writes to the FBO just like on Android, But I am not sure about the latter.

I'd be happy with a solution which lets the storyboard last until the user project starts,which is just what the boot splash logo is for and which currently doesn't work.

**Pull request merged**